src/python_testing/TC_FAN_3_3.py

```diff
@@ -0,0 +1,94 @@
+    ROCK_ROUND = 0x04
+
+    async def read_fc_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.FanControl
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_rock_support(self, endpoint):
+        return await self.read_fc_attribute_expect_success(endpoint, Clusters.FanControl.Attributes.RockSupport)
+
+    async def read_rock_setting(self, endpoint):
+        return await self.read_fc_attribute_expect_success(endpoint, Clusters.FanControl.Attributes.RockSetting)
+
+    async def write_rock_setting(self, endpoint, rock_setting):
+        result = await self.default_controller.WriteAttribute(self.dut_node_id, [(endpoint, Clusters.FanControl.Attributes.RockSetting(rock_setting))])
+        asserts.assert_equal(result[0].Status, Status.Success, "RockSetting write failed")
+
+    @async_test_body
+    async def test_TC_FAN_3_3(self):
+        if not self.check_pics("FAN.S.F02"):
+            logger.info("Test skipped because PICS FAN.S.F02 is not set")
+            return
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.print_step(1, "Commissioning, already done")
+
+        self.print_step(2, "Read from the DUT the RockSupport attribute and store")
+        rock_support = await self.read_rock_support(endpoint=endpoint)
+
+        if rock_support & self.ROCK_LEFT_RIGHT:
+            self.print_step(3, "RockLeftRight is supported, so write 0x01 to RockSetting")
+            await self.write_rock_setting(endpoint=endpoint, rock_setting=self.ROCK_LEFT_RIGHT)
+            time.sleep(1)
+
+            self.print_step(4, "Read from the DUT the RockSetting attribute")
+            rock_setting = await self.read_rock_setting(endpoint=endpoint)
+            asserts.assert_equal(rock_setting, self.ROCK_LEFT_RIGHT, "RockSetting is not 0x01")
+
+        if rock_support & self.ROCK_UP_DOWN:
+            self.print_step(5, "RockUpDown is supported, so write 0x02 to RockSetting")
+            await self.write_rock_setting(endpoint=endpoint, rock_setting=self.ROCK_UP_DOWN)
+            time.sleep(1)
+
+            self.print_step(6, "Read from the DUT the RockSetting attribute")
+            rock_setting = await self.read_rock_setting(endpoint=endpoint)
+            asserts.assert_equal(rock_setting, self.ROCK_UP_DOWN, "RockSetting is not 0x02")
+
+        if rock_support & self.ROCK_ROUND:
+            self.print_step(7, "RockRound is supported, so write 0x04 to RockSetting")
+            await self.write_rock_setting(endpoint=endpoint, rock_setting=self.ROCK_ROUND)
+            time.sleep(1)
+
+            self.print_step(8, "Read from the DUT the RockSetting attribute")
+            rock_setting = await self.read_rock_setting(endpoint=endpoint)
+            asserts.assert_equal(rock_setting, self.ROCK_ROUND, "RockSetting is not 0x04")
+
+        self.print_step(9, "Write RockSetting to 0x00")
```

Comment:
This doesn't appear in the test plan. It's correct though. ie, test plan bug.

Reply:
Will raise bug and link here.